lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -361,8 +377,52 @@ export class PublicClientApplication {
         // block the reload if it occurred inside a hidden iframe
         BrowserUtils.blockReloadInHiddenIframes();
 
-        // Send request to renew token. Auth module will throw errors if token cannot be renewed.
-        return await this.authModule.getValidToken(tokenRequest);
+        try {
+            // Send request to renew token. Auth module will throw errors if token cannot be renewed.
+            return await this.authModule.getValidToken(tokenRequest);
+        } catch (e) {
+            const isServerError = e instanceof ServerError;
+            const isInvalidGrantError = (e.errorCode === BrowserConstants.INVALID_GRANT_ERROR);
+            if (isServerError && isInvalidGrantError) {
+                const silentRequest: AuthenticationParameters = {
+                    scopes: tokenRequest.scopes,
+                    resource: tokenRequest.resource,
+                    extraQueryParameters: tokenRequest.extraQueryParameters,
+                    authority: tokenRequest.authority,
+                    correlationId: tokenRequest.correlationId,
+                    prompt: PromptValue.NONE,
+                    account: tokenRequest.account,
+                    sid: tokenRequest.sid,
+                    loginHint: tokenRequest.loginHint
+                };
```

Comment:
```js
const silentRequest = {
  ...tokenRequest,
  prompt: PromptValue.NONE
};
```

Reply:
Or do you only want these specific properties?

Reply:
If forceRefresh exists on `tokenRequest` does it get copied to the `silentRequest`?

Reply:
Never mind, I have made this change.